### ISSUE TITLE
Suddenly worsened in the middle of training

### ISSUE THREAD

**Issue by WrongMalik** [nianticlabs/monodepth2#344]
Hi,
I successfully ran the code on other machines, but the training results are a bit strange.
The parameters are set as follows, the others keep the default
> --model_name mono_model --png --num_epochs 30

My GPU is NVIDIA GeForce GTX 1080 Ti, CUDA=0.9.0, python=3.6.6, torch=0.4.1, torchvision=0.2.1, opencv=3.3.1.11

Below is the program output. The program was killed in the 29th epoch, but it did not affect the overall result.
> Training model named:
   mono_model
Models and tensorboard events files are saved to:
   ../data/yiw_logs/
Training is using:
   cuda
Using split:
   eigen_zhou
There are 39810 training items and 4424 validation items
Training
epoch   0 | batch      0 | examples/s:   5.4 | loss: 0.15755 | time elapsed: 00h00m21s | time left: 00h00m00s
epoch   0 | batch    250 | examples/s:  15.4 | loss: 0.14208 | time elapsed: 00h05m10s | time left: 34h13m20s
epoch   0 | batch    500 | examples/s:  13.8 | loss: 0.11805 | time elapsed: 00h09m32s | time left: 31h28m18s
epoch   0 | batch    750 | examples/s:  16.5 | loss: 0.13932 | time elapsed: 00h14m03s | time left: 30h51m51s
epoch   0 | batch   1000 | examples/s:  14.1 | loss: 0.12094 | time elapsed: 00h17m57s | time left: 29h29m24s
epoch   0 | batch   1250 | examples/s:  13.7 | loss: 0.10922 | time elapsed: 00h22m30s | time left: 29h29m22s
epoch   0 | batch   1500 | examples/s:  17.7 | loss: 0.11639 | time elapsed: 00h26m26s | time left: 28h48m10s
epoch   0 | batch   1750 | examples/s:  17.0 | loss: 0.08141 | time elapsed: 00h31m35s | time left: 29h24m55s
epoch   0 | batch   2000 | examples/s:  13.1 | loss: 0.12072 | time elapsed: 00h35m05s | time left: 28h30m34s
Training
epoch   1 | batch    683 | examples/s:  16.5 | loss: 0.11255 | time elapsed: 01h05m56s | time left: 26h14m22s
epoch   1 | batch   2683 | examples/s:  11.4 | loss: 0.10045 | time elapsed: 01h33m28s | time left: 24h16m52s
Training
epoch   2 | batch   1366 | examples/s:  13.3 | loss: 0.10931 | time elapsed: 02h01m15s | time left: 23h07m07s
Training
epoch   3 | batch     49 | examples/s:  12.8 | loss: 0.09068 | time elapsed: 02h29m34s | time left: 22h18m52s
epoch   3 | batch   2049 | examples/s:  13.6 | loss: 0.09970 | time elapsed: 02h57m15s | time left: 21h32m38s
Training
epoch   4 | batch    732 | examples/s:  16.3 | loss: 0.09274 | time elapsed: 03h25m04s | time left: 20h52m33s 
epoch   4 | batch   2732 | examples/s:  16.7 | loss: 0.06679 | time elapsed: 03h52m37s | time left: 20h14m11s 
Training
epoch   5 | batch   1415 | examples/s:  14.0 | loss: 0.09769 | time elapsed: 04h20m25s | time left: 19h39m16s 
Training
epoch   6 | batch     98 | examples/s:  14.2 | loss: 0.08676 | time elapsed: 04h48m10s | time left: 19h05m39s    
epoch   6 | batch   2098 | examples/s:  16.8 | loss: 0.11321 | time elapsed: 05h15m46s | time left: 18h32m30s    
Training
epoch   7 | batch    781 | examples/s:  15.4 | loss: 0.10928 | time elapsed: 05h43m32s | time left: 18h00m50s    
epoch   7 | batch   2781 | examples/s:  13.3 | loss: 0.09710 | time elapsed: 06h11m34s | time left: 17h30m32s    
Training
epoch   8 | batch   1464 | examples/s:  13.1 | loss: 0.12213 | time elapsed: 06h39m16s | time left: 16h59m43s    
Training
epoch   9 | batch    147 | examples/s:  16.9 | loss: 0.11782 | time elapsed: 07h07m00s | time left: 16h29m22s    
epoch   9 | batch   2147 | examples/s:  15.0 | loss: 0.12204 | time elapsed: 07h38m21s | time left: 16h06m59s    
Training
epoch  10 | batch    830 | examples/s:  14.9 | loss: 0.09638 | time elapsed: 08h10m30s | time left: 15h45m04s    
epoch  10 | batch   2830 | examples/s:   9.4 | loss: 0.12125 | time elapsed: 08h43m03s | time left: 15h22m45s    
Training
epoch  11 | batch   1513 | examples/s:  13.0 | loss: 0.11852 | time elapsed: 09h12m37s | time left: 14h54m31s    
Training
epoch  12 | batch    196 | examples/s:  14.9 | loss: 0.10688 | time elapsed: 09h40m19s | time left: 14h23m22s    
epoch  12 | batch   2196 | examples/s:  15.4 | loss: 0.10893 | time elapsed: 10h07m51s | time left: 13h52m19s    
Training
epoch  13 | batch    879 | examples/s:  16.1 | loss: 0.09884 | time elapsed: 10h35m35s | time left: 13h21m51s    
epoch  13 | batch   2879 | examples/s:  14.3 | loss: 0.11345 | time elapsed: 11h03m14s | time left: 12h51m31s    
Training
epoch  14 | batch   1562 | examples/s:  15.5 | loss: 0.09454 | time elapsed: 11h31m03s | time left: 12h21m35s    
Training
epoch  15 | batch    245 | examples/s:  15.1 | loss: 0.10162 | time elapsed: 11h58m57s | time left: 11h51m54s    
epoch  15 | batch   2245 | examples/s:  15.4 | loss: 0.10366 | time elapsed: 12h26m39s | time left: 11h22m11s    
Training
epoch  16 | batch    928 | examples/s:  17.5 | loss: 0.10269 | time elapsed: 12h54m32s | time left: 10h52m46s    
epoch  16 | batch   2928 | examples/s:  17.0 | loss: 0.10242 | time elapsed: 13h22m14s | time left: 10h23m18s    
Training
epoch  17 | batch   1611 | examples/s:  18.0 | loss: 0.10597 | time elapsed: 13h50m05s | time left: 09h54m05s    
Training
epoch  18 | batch    294 | examples/s:  19.5 | loss: 0.10446 | time elapsed: 14h17m55s | time left: 09h24m56s    
epoch  18 | batch   2294 | examples/s:  13.4 | loss: 0.09915 | time elapsed: 14h45m41s | time left: 08h55m50s    
Training
epoch  19 | batch    977 | examples/s:  14.0 | loss: 0.11154 | time elapsed: 15h13m36s | time left: 08h26m54s    
epoch  19 | batch   2977 | examples/s:  15.0 | loss: 0.09389 | time elapsed: 15h41m24s | time left: 07h57m58s    
Training
epoch  20 | batch   1660 | examples/s:  15.0 | loss: 0.10394 | time elapsed: 16h09m09s | time left: 07h29m05s    
Training
epoch  21 | batch    343 | examples/s:  13.7 | loss: 0.10519 | time elapsed: 16h36m51s | time left: 07h00m14s    
epoch  21 | batch   2343 | examples/s:  15.0 | loss: 0.10809 | time elapsed: 17h04m22s | time left: 06h31m23s    
Training
epoch  22 | batch   1026 | examples/s:  16.1 | loss: 0.10612 | time elapsed: 17h32m08s | time left: 06h02m42s    
epoch  22 | batch   3026 | examples/s:  14.2 | loss: 0.08656 | time elapsed: 17h59m45s | time left: 05h34m00s    
Training
epoch  23 | batch   1709 | examples/s:  13.1 | loss: 0.08813 | time elapsed: 18h27m26s | time left: 05h05m23s    
Training
epoch  24 | batch    392 | examples/s:  17.3 | loss: 0.09563 | time elapsed: 18h55m06s | time left: 04h36m49s    
epoch  24 | batch   2392 | examples/s:  13.6 | loss: 0.10738 | time elapsed: 19h22m45s | time left: 04h08m17s    
Training
epoch  25 | batch   1075 | examples/s:  13.7 | loss: 0.10045 | time elapsed: 19h50m28s | time left: 03h39m48s    
epoch  25 | batch   3075 | examples/s:  16.4 | loss: 0.09057 | time elapsed: 20h18m07s | time left: 03h11m21s    
Training
epoch  26 | batch   1758 | examples/s:  15.2 | loss: 0.09262 | time elapsed: 20h45m46s | time left: 02h42m56s    
Training
epoch  27 | batch    441 | examples/s:  12.9 | loss: 0.11443 | time elapsed: 21h13m31s | time left: 02h14m34s    
epoch  27 | batch   2441 | examples/s:  14.3 | loss: 0.08789 | time elapsed: 21h41m03s | time left: 01h46m12s    
Training
epoch  28 | batch   1124 | examples/s:  12.9 | loss: 0.10578 | time elapsed: 22h09m02s | time left: 01h17m54s    
epoch  28 | batch   3124 | examples/s:  15.3 | loss: 0.10685 | time elapsed: 22h38m54s | time left: 00h49m41s    
Training

The result is very strange, it suddenly worsened in the 7th epoch. The orange line is train, the blue line is val.
![da](https://user-images.githubusercontent.com/36250260/118364431-f72ee900-b5ca-11eb-938c-b805987efac6.png)
![de](https://user-images.githubusercontent.com/36250260/118363942-2cd2d280-b5c9-11eb-903b-b679bd7f8fff.png)
![loss](https://user-images.githubusercontent.com/36250260/118364434-fe55f700-b5ca-11eb-9234-6971e98356ad.png)

And the evaluation result of 20th epoch is bellow.
> abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
&   0.261  &   2.207  &   7.801  &   0.343  &   0.556  &   0.819  &   0.928  \\

I want to know the reason, sorry to bother you once again.

**Comment by WrongMalik** [comment]
I did another training with parameters：--model_name mono_model --png --num_epochs 30 ----learning_rate 1e-5
![1e-5da](https://user-images.githubusercontent.com/36250260/118369535-547c6780-b5d6-11eb-820f-87bbc24ec17a.png)
![1e-5de](https://user-images.githubusercontent.com/36250260/118369536-55ad9480-b5d6-11eb-9ac8-31ad839dedab.png)
![1e-5loss](https://user-images.githubusercontent.com/36250260/118369538-56462b00-b5d6-11eb-8ec9-27c807ff0073.png)

**Comment by mdfirman** [comment]
Hi – thanks for reporting this. I've not seen this happen before.

It looks like you're using a lower learning rate than we did though. Does it train correctly if you use `1e-4` like we did?

Also it would be worth looking at the images in the tensorboard – what do they look like? Thanks.

**Comment by WrongMalik** [comment]
Hi,

The learning rate of the first training is 1e-4, the second one is 1e-5. Images prediction results are poor.
![MD`QN_UXQ@}6V EOV$%DQO](https://user-images.githubusercontent.com/36250260/118608531-5f104a00-b7ec-11eb-92cf-6d5da65723cb.png)
![{6(})VQ8Y188NK5V1Q%0 D7](https://user-images.githubusercontent.com/36250260/118608539-60da0d80-b7ec-11eb-86f0-ee2781c54d98.png)


I ran the code on the third machine. It behaves as well as the paper. The difference between this experiment and the previous few times is that the experimental data this time is on the local, and the previous few times on the mounted mobile hard disk. Is the data reading affected? But there is no problem with the operation of writing logs and models to the mobile hard disk.

**Comment by mdfirman** [comment]
Thanks for reporting back.

This is very strange. It is learning someting, but seems to be learning an 'average' scene shape rather than the shape of each specific image.

I would definitely double check that the datasets are the same between the two machines. Could the local one be corrupted in some way? E.g. the filenames being wrong or out of order? Do the 'left' and 'right' images match when you look at them in the tensorboard?

**Comment by WrongMalik** [comment]
Hi,

Sorry to reply to you after so long, I have been busy finishing my undergraduate thesis recently. 

I checked the pictures in tensorboard, the left and right pictures are corresponding. At the same time, I also checked the path name and MD5 value of each picture in the two data sets, and they are exactly the same.

Is this problem related to the mounting status of the mobile hard disk? I have a related guess as to whether it will affect the reading of data when the portable hard disk is temporarily dropped. But every log write of the program is normal.

**Comment by mdfirman** [comment]
I have to say this is very strange and I don't know exactly what is going on. If images can't be found (e.g. by a temporary drop of disk) in general the training should die, rather than be corrupted.

Perhaps one final thing to try would be to train with `--num_workers 0`. This will be much slower to train than normal, but it means any problems in loading images from disk should be more likely to be printed to terminal (rather than hidden inside the threads).

**Comment by mdfirman** [comment]
Did you have any more success with this debugging?

**Comment by porwalnaman01** [comment]
Hi, I am also facing similar issue, the loss isn't converging while training. Have you found the solution?
I am using PyTorch 1.8.0, torchvision 0.9.0 on CUDA 11.1.
Thanks in advance!

**Comment by mdfirman** [comment]
@porwalnaman01  – thanks for reporting in here. If you are using pytorch 1.8.0 I wouldn't expect this to work. The repo only works with the version of pytorch we specify, as some behaviours have changed between the two versions.

Thanks,

Michael

**Comment by jamesheatonrdm** [comment]
Hi, I am also having issues with this. I am using the correct version of PyTorch.
